### PR TITLE
Nextflow VEP: support Seqera platform

### PR DESCRIPTION
**Pull request by nuno-agostinho**
ENSVAR-5853: Changes to organisation of Nextflow VEP files in order to support Seqera platform

## Changelog

* [Accept headerless VCF files](https://github.com/Ensembl/ensembl-vep/pull/1724/commits/15cd3a9233d1a9c5fb8396eac4c1ff29ff79ab51)
    - If a VCF has no header, process as a normal text file (like done for non-VCF input files)
* Add `main.nf` and `nextflow.config` to root directory
* Move `nextflow.config` to `nextflow` directory
* Rename the `nf-modules` directory to `modules`
* Add Nextflow schema
    - Move pipeline parameters to `nextflow.config`
    - Add sections of [nf-core config template](https://github.com/nf-core/tools/blob/master/nf_core/pipeline-template/nextflow.config) to `nextflow.config`
* Updated `README.md`

## Testing

Run pipeline using:
```
nextflow run nuno-agostinho/ensembl-vep -r nextflow/seqera-platform
```

After the pipeline is merged to main and release branches, it can be run properly:
```
nextflow run Ensembl/ensembl-vep
nextflow run Ensembl/ensembl-vep -r release/113
```

### CODE REVIEW

File: nextflow/nextflow.config [hunk 1]
```diff
@@ -1,30 +1,130 @@
...
   slurm {
     process.executor = 'slurm'
   }
-}
 
-singularity {
-  enabled = true
-  autoMounts = true
+  conda {
+    conda.enabled          = true
```
Comment (veidenberg) [inline nextflow/nextflow.config:39]: Just checking, can VEP bypass Anaconda?

File: nextflow/workflows/run_vep.nf [hunk 2]
```diff
@@ -87,15 +73,16 @@ workflow vep {
     inputs |
       branch {
         index: it.file =~ '\\.(tbi|csi)$'
-        registry: it.file =~ '\\.registry$'
-        ini: it.file =~ '\\.ini$'
-        vcf: it.file =~ '\\.vcf(.gz)?$'
+        ignore: it.file =~ '\\.(ini|registry|config)$'
+        vcf_with_header:
+          ( it.file =~ '\\.vcf$' && it.file.readLines()[0][0] =~ '^#' ) ||
```
Comment (nakib103) [inline nextflow/workflows/run_vep.nf:78]: @nuno-agostinho,

To run a VCF file through the `runVEPonVCF` we should ensure the file have both fileformat and the header line -
```
##fileformat=VCFv4.2
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO
```

Can we check if both of these lines are present?

**Review by nakib103 — CHANGES_REQUESTED** [review]

File: nextflow/workflows/run_vep.nf (hunk 2 above)
Comment (nuno-agostinho) [reply nextflow/workflows/run_vep.nf]: Thanks for the suggestion! Included in new commit: https://github.com/Ensembl/ensembl-vep/pull/1724/commits/b8d44e974caa81a2538b2002b82df789a2485908

**Review by nakib103 — APPROVED** [review]
Looks good!~

**Pull request merged**